plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -52,6 +52,25 @@ export function castingCompare(
     b: any,
     operator: PropertyOperator.Exact | PropertyOperator.IsNot | PropertyOperator.LessThan | PropertyOperator.GreaterThan
 ): boolean {
+    // Check null case first
```

Comment:
Can you add the rationale from https://github.com/PostHog/posthog/issues/12893#issuecomment-1323645354 here? Would greatly help future travelers

Reply:
Makes sense!